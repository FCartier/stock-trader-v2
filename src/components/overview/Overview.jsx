import React from "react";
import { SectionHeader } from "../reusables/SectionHeader"

const Overview = ({overview}) => {
  return Object.keys(overview).length ? (
    <div>
        <div className="overview">
          <SectionHeader title="COMPANY OVERVIEW"/>
          <div className="overview-content">
            <h2>{overview.companyName}</h2>
            <p>{overview.website}</p>
            <p>{overview.description}</p>
          </div>
        </div>
    </div>
  ) : null
};

export default Overview;
