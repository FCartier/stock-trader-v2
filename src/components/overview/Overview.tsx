import * as React from "react";
import { SectionHeader } from "../reusables/SectionHeader"

export interface IOverviewProps {
  overview?: {
      companyName: string;
      website: string;
      description: string;
  };
}

const Overview: React.SFC<IOverviewProps> = ({overview}) => {
  return !(typeof overview === "undefined" || overview.companyName === "") ? (
    <div>
      <div className='overview'>
        <SectionHeader title="COMPANY OVERVIEW" />
        <div className='overview-content'>
          <h2>{overview.companyName}</h2>
          <p>{overview.website}</p>
          <p>{overview.description}</p>
        </div>
      </div>
    </div>
  ) : null;
};

export default Overview;
