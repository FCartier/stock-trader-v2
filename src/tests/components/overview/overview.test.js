import React from 'react';
import Overview from '../../../components/overview/Overview';
import renderer from 'react-test-renderer';
import testData from './overview.data.json'


describe("overview component", () => {
  test('overview component should display all data fields', () => {
    const component = renderer.create(
      <Overview overview={testData}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
