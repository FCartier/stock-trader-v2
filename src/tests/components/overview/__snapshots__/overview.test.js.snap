// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overview component overview component should display all data fields 1`] = `
<div>
  <div
    className="overview"
  >
    <div
      className="category-header"
    >
      <h3>
        COMPANY OVERVIEW
      </h3>
    </div>
    <div
      className="overview-content"
    >
      <h2>
        American Assets Trust Inc.
      </h2>
      <p>
        http://www.americanassetstrust.com
      </p>
      <p>
        American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
      </p>
    </div>
  </div>
</div>
`;

exports[`overview component overview should display all data fields 1`] = `
<div>
  <div
    className="overview"
  >
    <div
      className="category-header"
    >
      <h3>
        COMPANY OVERVIEW
      </h3>
    </div>
    <div
      className="overview-content"
    >
      <h2>
        American Assets Trust Inc.
      </h2>
      <p>
        http://www.americanassetstrust.com
      </p>
      <p>
        American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
      </p>
    </div>
  </div>
</div>
`;

exports[`overview should display all data fields 1`] = `
<div>
  <div
    className="overview"
  >
    <div
      className="category-header"
    >
      <h3>
        COMPANY OVERVIEW
      </h3>
    </div>
    <div
      className="overview-content"
    >
      <h2>
        American Assets Trust Inc.
      </h2>
      <p>
        http://www.americanassetstrust.com
      </p>
      <p>
        American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
      </p>
    </div>
  </div>
</div>
`;
